### ISSUE TITLE
I am very interested in your tools. Can you elaborate on how to use your tools?

### ISSUE THREAD

**Issue by chilatiaobushanghuo** [jsoref/spelling#27]

**Comment by jsoref** [comment]
There are four things that kind of work together (each of which can be more or less used individually):
* [spelling](https://github.com/jsoref/spelling) -- misc tools for fixing spelling errors in repositories -- esp. [d](../tree/HEAD/d), [hesort](../tree/HEAD/hesort), [hgfileexts](../tree/HEAD/hgfileexts), [hgfilesample](../tree/HEAD/hgfilesample), [hgmv](../tree/HEAD/hgmv), [r](../tree/HEAD/r), [rs](../tree/HEAD/rs), [s](../tree/HEAD/s), and [w](../tree/HEAD/w). -- Note that the versions I have locally may be newer than the versions linked.
* [spell-check-this](https://github.com/check-spelling/spell-check-this) -- template configuration for [actions/check-spelling](https://github.com/marketplace/actions/check-spelling)
* [check-spelling](https://github.com/check-spelling/check-spelling) -- the check-spelling action (based on `spelling`'s [w](../tree/HEAD/w))
* [check-spelling/wiki](https://github.com/check-spelling/check-spelling/wiki) -- scripts that are fairly specific to the action and patterns used by the action

I'm slowly evolving the process. Initially, there was just `spelling`. And the workflow was as described in [workflows.md](../tree/HEAD/workflows.md).

These days, my process is more like this:
1. Fork a repo.
2. Use `empty-remote` (which I'll probably add to this repo eventually, it just gets rid of all the other branches/tags so that my clones are smaller) on the fork.
3. Clone the fork.
4. Copy in [spell-check-this/.github](https://github.com/check-spelling/spell-check-this/tree/master/.github).
5. Import [Looking-for-items-locally](https://github.com/check-spelling/check-spelling/wiki/Looking-for-items-locally).
6. Use [hgfileexts](../blob/HEAD/hgfileexts) and [hgfilesample](../blob/HEAD/hgfilesample) to select file extensions to ignore (there's an `hgfileignore` variant that I'll push at some point) -- ignored items are added to `.github/actions/spelling/excludes.txt` (see [Configuration Examples: excludes](https://github.com/check-spelling/check-spelling/wiki/Configuration-Examples:-excludes)).
7. Use `echo "$files"|tree --fromfile=.` to identify other things to ignore (especially `(?:^|/)vendor/`, `(?:^|/)package(?:-lock|)\.json$`, sometimes localization directories, e.g. `/[^B][^/]*\.lproj/`, `\.min\.` (minified resources).
8. Generate `patterns.txt` often based on the [Configuration Examples: examples](https://github.com/check-spelling/check-spelling/wiki/Configuration-Examples:-patterns), but sometimes writing repo specific patterns and sometimes updating the examples.
9. Use `search` (from `Looking-for-items-locally`) to generate a list of tokens to consider.
10. Use `review` (from `Looking-for-items-locally`) to review the output of `search`.
11. Iterate through steps 5-10.
12. Once I've excluded enough files and patterns, I checkpoint my spelling work -- `git add .github/; s 'xxx xxx'` (the commit message is eventually replaced with `Add check-spelling/check-spelling` or similar which I copy+paste near the end).
13. And send the remaining tokens to a Google Sheet which has a couple of columns (I fill the same values into `Original` and `Replacement` initially; `review` compares the two and blanks when they differ...
    Command | rs | Original | Replacement | review |
    --|--|--|--|--
     | | | `alpah` | `alpah` | `?`
     | | | `absolutelyfabulous` | `absolutelyfabulous` | `?`
     | | | | |
     | | | | |
     | | | | |


14. I select the `Replacement` column and ask [Google Sheets](http://sheets.google.com/) to run a [Spell check](https://spreadsheetpoint.com/spell-check-google-sheets/). When it's done, I have something like:
    Command | rs | Original | Replacement | review |
    --|--|--|--|--
    `rs "alpah" alpha` | `rs` | `alpah` | `alpha` |
     | | | `absolutelyfabulous` | `absolutelyfabulous` | `?`
     | | | | |
     | | | | |
     | | | | |

15. I use the `review` column to hide either the unreviewed or the reviewed things (usually I'll do extra passes after Google's checker finishes).

16. Then I sort things based on the replacement (this isn't as easy as it sounds, I actually copy the Command column to the right edge of the sheet, split the result on ` `, and then do some more magic to select the replacement.
    * [rs](../blob/HEAD/rs) takes 3 arguments: `original`, `replacement`, `commit term` -- the third item is optional -- the second is used when the third is omitted.
    * Sometimes I'm replacing `sease` with `cease` and the sorting is significant.
    * Sometimes I'll be replacing `foodWaist` with `foodWaste` and the corrected word is `waste` and should be sorted accordingly.

17. Once this is done, I'll copy the `Command` column back to a terminal and let it run...
18. If things work properly, I review the work (typically `git log main..spelling --word-diff --oneline --graph -U0`).
19. Often I'll have to use [hgmv](../blob/HEAD/hgmv) in a `git rebase` to deal w/ file renames because `rs` doesn't currently handle that.
20. Then I use `git rebase -i main` and `hesort .git/rebase/git-rebase-todo` (or something like that) to fix up and merge related words.
21. Then I use `reverse-line < words | sort -uf > .github/actions/spelling/expect.txt`, add, and commit it, and rebase...
22. Then there's some dancing to split the spell-check commit off the stack, send it, and the spelling branch to github, and an extra combined branch to validate things -- at this point, I have a `/compare/spelling-with-spell-check...spell-check` clipboard-ish item that I use (because the GitHub PR chooser loves comparing a fork to its upstream whereas I don't want to do that).
23. Once things work, I create PRs by filling out a [PR Template](https://github.com/check-spelling/check-spelling/wiki/PR-Template).

----
The action itself can be used with just `push` if you don't take PRs, or with `schedule` (until my next release which will hopefully be today) or `pull_request_target` (with the next release) to respond to incoming PRs. There are a couple of projects which have started using it and seem fairly happy with it.

In general, if I've already done the work, it isn't a problem for me to offer the `spell-check` action configuration. Even if I haven't, it isn't a big deal. The only important thing is to make sure that the tree is green before merging, otherwise, people tend to get very confused and frustrated -- this isn't specific to this action, it applies to any code checking action -- it's generally frustrating making a PR and getting an error that's unrelated to the PR because the main branch is failing.

If a project is too big to do all at once, one can use `only` or `excludes` to focus on a smaller subset of the repository and then expand in a branch and relax the exclusions until it's more completely covered. -- There's one project which is only using the action for documentation (i.e. excluding code).

Some people have set up the action independently, it isn't too hard.

----

I'm not particularly proficient at managing my own repositories (I'm used to being a contributor elsewhere) and am only now learning how to use the `watch` feature. So please accept my apologies for not noticing this issue sooner.

That said, at the time you filed it, the answer was entirely contained within [workflows.md](../tree/HEAD/workflows.md) and linked from the [Overview](../#overview).
